frontend/src/lib/components/LemonCalendar/LemonCalendarRange.tsx

```diff
@@ -0,0 +1,146 @@
+import { LemonCalendar, LemonCalendarProps } from 'lib/components/LemonCalendar/LemonCalendar'
+import React, { useEffect, useState } from 'react'
+import { dayjs } from 'lib/dayjs'
+import { LemonButton } from 'lib/components/LemonButton'
+import { IconClose } from 'lib/components/icons'
+import { formatDateRange } from 'lib/utils'
+
+export interface LemonCalendarRangeProps {
+    value?: (string | null)[]
+    onChange: (date: string[]) => void
+    months?: number
+    getLemonButtonProps?: LemonCalendarProps['getLemonButtonProps']
+    onClose?: () => void
+}
+
+export function LemonCalendarRange({ value, onChange, onClose, months }: LemonCalendarRangeProps): JSX.Element {
+    // Keep a sanitised and cached copy of the selected range
+    const [valueStart, valueEnd] = [
+        value?.[0] ? dayjs(value[0]).format('YYYY-MM-DD') : null,
+        value?.[1] ? dayjs(value[1]).format('YYYY-MM-DD') : null,
+    ]
+    const [[rangeStart, rangeEnd], setRange] = useState([valueStart, valueEnd])
+
+    // Track if the last change was on the range's start or end
+    const [lastChanged, setLastChanged] = useState('end' as 'start' | 'end')
```

Comment:
I don't find this _totally_ intuitive, because this way I have to kind of go out of my way to change BOTH the start and end:

![2022-08-31 13 47 06](https://user-images.githubusercontent.com/4550621/187671889-3c76ffe8-cda5-4de3-b914-25f9da6ce7aa.gif)

A model that I've found to work great is just switching between start and end selection each time. Then for me there's no need to figure out that I have to click the _other_ start/end to switch from moving around end/start to start/end (see, this is complex). Google Flights' picker does it nicely: https://www.google.com/flights

Reply:
That's how it was supposed to work, but seems like something broke. Added a test to catch this behaviour.